docs/.vitepress/versions.ts

```diff
@@ -1,7 +1,58 @@
+import { execSync } from 'node:child_process';
+import * as semver from 'semver';
 import { version } from '../../package.json';
 
-export const currentVersion = `v${version}`;
+export const NEXT_TEXT = 'next';
 
+function pickLatest(version1: string, version2: string): string {
+  return semver.compare(version1, version2) > 0 ? version1 : version2;
+}
+
+function readBranchName(): string {
+  return (
+    execSync('git branch --show-current').toString('utf8').trim() || NEXT_TEXT
+  );
+}
+
+function readOtherLatestReleaseTagNames(): string[] {
+  const currentMajorVersion = semver.major(version);
+  const latestReleaseTagNames = execSync('git tag -l')
+    .toString('utf8')
+    .split('\n')
+    .filter((tagName) => tagName.startsWith('v'))
```

Comment:
I think it's more safe to use `valid` here

```suggestion
    .filter((tag) => semver.valid(tag))
```